internal/config/config.go

```diff
@@ -0,0 +1,131 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/fsnotify/fsnotify"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+)
+
+// validateDistributions checks that the condition that
+// two distributions with the same distribution ID MUST NOT share the same prefix.
+// or in other terms, every pair of id,prefix (Distribution) must be unique
+func validateDistributions(distributions distributionsMap) error {
+	uniqueMap := make(map[*Distribution]struct{})
+
+	for _, value := range distributions {
+		if _, ok := uniqueMap[value]; ok {
+			return fmt.Errorf("error parsing configuration: distribution value duplicated id: %s prefix: %s", value.ID, value.Prefix)
+		}
+
+		uniqueMap[value] = struct{}{}
+	}
+
+	return nil
+}
+
+func (c *Config) parse(data []byte) error {
+	config := initConfig()
+
+	if err := yaml.Unmarshal(data, config); err != nil {
+		return err
+	}
+
+	err := validateDistributions(config.Distributions)
+	if err != nil {
+		return err
+	}
+
+	for name, value := range config.Distributions {
+		c.distributions.Set(name, value)
+	}
+	for _, name := range c.distributions.Names() {
+		if _, ok := config.Distributions[name]; !ok {
+			c.distributions.Delete(name)
+		}
+	}
+
+	for name, value := range config.Entitlements {
+		c.entitlements.Set(name, value)
+	}
+	for _, name := range c.entitlements.Names() {
+		if _, ok := config.Entitlements[name]; !ok {
+			c.entitlements.Delete(name)
+		}
+	}
+
+	return nil
+}
+
+func (c *Config) Load(file string) error {
```

Comment:
This should by unexported.

Reply:
any specific reason? Is it because the intention is to use Watch?